StringTokenFormatter/Containers/NonLockingLazy.cs

```diff
@@ -22,10 +22,16 @@ public T Value
         {
             get
             {
+                //Defensive copy
+                var cachedcreator = creator;
```

Comment:
I don't understand this line, if IsValueCreated is true then we'll have created the local variable for nothing.

Reply:
In a multithreaded environment, it is possible to have lop-sided multithreadded calls.  If we just had this code:

````
if (!IsValueCreated)
                {
                    CreatedValue = creator();
                    IsValueCreated = true;
                    //Null this out so we don't keep captured values around.
                    creator = null;
                }
                return CreatedValue;
````

Consider what would happen with this order of operations:
Thread1: Check if value is created.  It isn't so we'll enter the IF.
Thread2: Check if value is created.  It isn't so we'll enter the IF.
Thread1:  CreatedValue = creator();
Thread1:  IsValueCreated = true;
Thread1:  creator = null;
Thread1:  return CreatedValue;
Thread2: CreatedValue = creator(); //CRASH!

Thread 2 causes a crash because creator() got nulled out by thread1 before it could be invoked.
The defensive copy prevents that unlikely occurance in tightly wound multithreaded environments.